tomviz/CentralWidget.cxx

```diff
@@ -257,14 +271,26 @@ CentralWidget::CentralWidget(QWidget* parentObject, Qt::WindowFlags wflags)
 
   this->EventLink->Connect(chart, vtkCommand::CursorChangedEvent, this,
                            SLOT(histogramClicked(vtkObject*)));
+
+  this->Worker->start();
+  this->HistogramGen->moveToThread(this->Worker);
+  this->connect(this->HistogramGen,
+     SIGNAL(histogramDone(vtkSmartPointer<vtkImageData>,vtkSmartPointer<vtkTable>)),
```

Comment:
So the answer to several of your questions is here.  The worker thread is stared and then the HistogramGen object is moved to the other thread.  By Qt's threading rules the main (GUI) thread should not mess with its data anymore and slots triggered by Qt signals (or invokeMethod) will be run on the thread that owns the object.